testing/test_capture.py

```diff
@@ -515,18 +515,33 @@ def test_hello(capfdbinary):
         reprec.assertoutcome(passed=1)
 
     def test_capsysbinary(self, testdir):
-        reprec = testdir.inline_runsource(
+        p1 = testdir.makepyfile(
             """\
             def test_hello(capsysbinary):
                 import sys
+
                 # some likely un-decodable bytes
                 sys.stdout.buffer.write(b'\\xfe\\x98\\x20')
+
                 out, err = capsysbinary.readouterr()
                 assert out == b'\\xfe\\x98\\x20'
                 assert err == b''
+
+                # handles writing strings
```

Comment:
It would be nice to include a link to the original issue here, so that whoever investigates if something goes wrong can find all the design discussion quickly 😄

Reply:
Good idea - updated. I also made the test a bit stricter.